api/v1alpha1/mariadb_replication_types.go

```diff
@@ -198,27 +198,18 @@ type Replication struct {
 	Enabled bool `json:"enabled,omitempty"`
 }
 
-// SetDefaults sets reasonable defaults.
-func (r *Replication) SetDefaults(mdb *MariaDB, env *environment.OperatorEnv) error {
-	if r.GtidStrictMode == nil {
-		r.GtidStrictMode = ptr.To(true)
-	}
-	if reflect.ValueOf(r.InitContainer).IsZero() {
-		r.InitContainer = InitContainer{
-			Image: env.MariadbOperatorImage,
-		}
+func (r *Replication) GetReplica() *ReplicaReplication {
```

Comment:
Instead of providing methods for safely accessing both primary and replica, we should change the primary and replica types to be structs like we did in Galera. That way we avoid the risk of dereferencing a nil pointer.